Lib/ufo2ft/__init__.py

```diff
@@ -172,6 +172,7 @@ def compileTTF(
     rememberCurveType=True,
     removeOverlaps=False,
     overlapsBackend=None,
+    flattenComponents=None,
```

Comment:
In the preprocessor classes, `flattenComponents` parameter is a boolean True/False; whereas in here it is `None`.
Does `None` carry any additional meaning? I don't see anywhere that you are checking for `None`.

We need to decide whether we want this to be a True/False only kind of parameter (where False _never_ runs flattenComponents, even when the filter is specified in the ufo.lib, and True _always_ does), or we want to have a third state `None` which means "only do flatten if the filter is specified in the ufo.lib". So the compile parameter would be used to override the lib filter.

I think it's better to have a True/False only option that ignores the lib filter; the worst it can happen is the ufo lib contains the filter already and we run it twice (but it's idempotent).

Reply:
No, that should just be a False.